admin/proc-using-aws-cognito-auth-provider.adoc

```diff
@@ -0,0 +1,203 @@
+[id='proc-using-aws-cognito-auth-provider_{context}']
+= Using Amazon Cognito as an authentication provider in {product}
+
+In this section, Amazon Cognito is an AWS service for adding an authentication layer into {product-short}. You can sign in directly to the {product-short} using a user pool or fedarate through a third-party identity provider.
+
+Although Amazon Cognito is not part of the core authentication providers for the Developer Hub, it can be integrated using the generic OpenID Connect (OIDC) provider.
+
+You can configure your {product-short} in both deployments Helm Chart and Operator-backed.
+
+.Prerequisites
+
+* You have a User Pool or you have created a new one. For more information about user pools, see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools.html?icmpid=docs_cognito_console_help_panel[Amazon Cognito user pools] documentation.
++
+[NOTE]
+====
+Ensure that you have note the AWS region where the user pool is located and the user pool ID.
+====
+
+* You have created an App Client within your user pool for integrating the hosted UI. For more information, see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-integration.html#cognito-user-pools-create-an-app-integration[Setting up the hosted UI with the Amazon Cognito console].
+
+Helm deployment::
++
+--
+.Procedure
+
+. Edit or create your custom `app-config-rhdh` ConfigMap as follows:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+    # --- Truncated ---
+    app:
+      title: Red Hat Developer Hub
+
+    signInPage: oidc
+    auth:
+      environment: production
+      session:
+        secret: ${AUTH_SESSION_SECRET}
+      providers:
+        oidc:
+          production:
+            clientId: ${AWS_COGNITO_APP_CLIENT_ID}
+            clientSecret: ${AWS_COGNITO_APP_CLIENT_SECRET}
+            metadataUrl: ${AWS_COGNITO_APP_METADATA_URL}
+            callbackUrl: ${AWS_COGNITO_APP_CALLBACK_URL}
+            scope: 'openid profile email'
+            prompt: auto
+----
+
+. Edit or create your custom `secrets-rhdh` Secret as follows:
```

Comment:
We should maybe write that this is just a template, and that the values for the keys need to be adjusted.

Reply:
Done